### ISSUE TITLE
CamelCase option to allow snake_case in object literals

### ISSUE THREAD

**Issue by joemaller** [eslint/eslint#1919]
There are situations where third-party code or services are configured by calling a function with an object literal. Those objects are flagged the camelCase rule:

```javascript
var snakes = require('somesnakelib');  // not my code
snakes({snake_prop: true});  // ESLint: "Identifier 'snake_prop' is not in camel case."
```

JSCS dealt with this by adding an [`ignoreProperties`](http://jscs.info/rules.html#requirecamelcaseoruppercaseidentifiers) option to their CamelCase rule. I like enforcing names and would rather not disable the CamelCase rule to use code that I don't control without errors. 

This was touched on in #656,  #672, #674 and #1550, but object literals are still being flagged.

**Comment by nzakas** [comment]
I'm not sure if ignoring all properties is what you want. For instance you'd also end up ignoring:

```js
var person = {
    say_name: function(){}
};
```

This is clearly a case where you would want to flag `say_name`. So maybe what makes the most sense is to add a flag to that ignores properties unless the object contains a method?

**Comment by joemaller** [comment]
True, but if it's just as likely that someone might use a function to fetch data to submit to an external service that used snake_case properties in their configuration object. 

```javascript
var person = fetchPerson({
      first_name: function(user) {
        return user.firstName;
      });
```
Sure that's a contrived example, but it's not difficult to imagine some logic in there. (though, gah.)

Anyway, thanks for considering this.

**Comment by nzakas** [comment]
Hmmm... Maybe we just need a new option `properties` that can be "always" (default), " never", "nonmethods".

**Comment by doberkofler** [comment]
This would be great addition to this rule and allow me to get rid of a ton of disable comments

**Comment by fflorent** [comment]
:+1:

**Comment by Tiagojdferreira** [comment]
Was the nonmethods property dropped ?

**Comment by OmgImAlexis** [comment]
@nzakas any updates on this?

I have the code below yet I'm still getting an error.

````
var client = tumblr.createClient({
    'consumer_key': config.get('tumblr:token'),
    'consumer_secret': config.get('tumblr:tokenSecret'),
    'token': tokenSet.token,
    'token_secret': tokenSet.tokenSecret
});
````

I'm using @sindresorhus's [xo linter](https://github.com/sindresorhus/xo/).

````
    "xo": {
        "space": 4,
        "rules": {
            "space-before-function-paren": 0,
            "camelcase": [
                2, {
                    "properties": "nonmethods"
                }
            ]
        }
    }
````

**Comment by OmgImAlexis** [comment]
@nzakas any reason a commit was merged but without the "nonmethods" option?

**Comment by MarkAPhillips** [comment]
Is the nonmethods property going to be included? Just has this issue in our code and this property setting would solve this issue.
